### ISSUE TITLE
current_person() returns null on authenticate

### ISSUE THREAD

**Issue by kirkbross** [graphile/postgraphile#1422]
### Summary

I'm trying to get a handle on the [current_person() function](https://www.graphile.org/postgraphile/postgresql-schema-design/#using-the-authorized-user) which (I believe) should return the 'person' identified by the token. For me, I get the token back, but `currentPerson {}` is always null. 

### Additional context

My thinking is that you send up an email and password and the authenticate function checks them against `person_account` —if an email match is found, it generates and returns a jwtToken. Now, I can certainly parse the token, extract the person_id, then use that to query `personByRowId` and get whatever "person" columns I want (fullName, avatarUrl, etc.)... but why can't authenticate send that data back with the token on the first trip? Or maybe it does and I'm missing something? Disclaimer: I'm an advanced novice so that is very possible.

In graphiql, the `authenticate` mutation has `query { currentPerson { fullName, avatarUrl, etc }}` available, but it always returns null. Assuming my database is identical to your schema design tutorial, my general pg server config is thus:
```
const database = postgres://mydb_postgraphile:MyPass!@localhost:5432/mydb_db

const options = {
  classicIds: true,
  enhanceGraphiql: true,
  exportGqlSchemaPath: `${__dirname}/pgSchema.graphql`,
  graphiql: true,
  jwtPgTypeIdentifier: 'mydb.jwt_token',
  jwtPublicKey: '',
  jwtSecret: 'mysecret',
  pgDefaultRole: 'mydb_anonymous',
};

// mutation

const mutation = graphql`
  mutation AuthenticateMutation($input: AuthenticateInput!) {
    authenticate(input: $input) {
      jwtToken // yep, lots of token love
      query { // available according to graphiql
        currentPerson { // nope, no person love
          fullName
          avatarUrl
        }
      }
    }
  }
`;

```
### TLDR;  

Is the `authenticate` mutation meant to return the `currentPerson {}` object in the same round trip?

**Comment by benjie** [comment]
That won’t work because none of the claims are set in the postgres transaction yet - they get set by you passing the token as a header which you obviously can’t do until you have the token. However it does work in the (more complex) implementation in Graphile Starter, where login is implemented in a plugin and then can set these settings before continuing with resolution.

**Comment by kirkbross** [comment]
@benjie Many thanks, sir. That was kind of what I suspected, but I wanted to confirm before I created unnecessary code. I'm going to fire up the Starter and do some deep learning. That looks like exactly what I need. Merry Christmas and stay safe.
